vishwamai/training.py

```
@@ -115,6 +127,18 @@ def __init__(
                 num_workers=2,
                 pin_memory=True
             )
+        
+        # Initialize metrics
+        self.best_val_loss = float('inf')
+        self.patience_counter = 0
+        
+        # Initialize wandb for monitoring
+        try:
+            import wandb
+            wandb.init(project="vishwamai", config=kwargs)
```

Ruff (0.8.2)

Undefined name `kwargs`

(F821)